packages/framework/tests/Feature/AutomaticNavigationConfigurationsTest.php

```diff
@@ -0,0 +1,292 @@
+            new MarkdownPage('first', ['navigation.priority' => 1]),
+            new MarkdownPage('second', ['navigation.priority' => 2]),
+            new MarkdownPage('third', ['navigation.priority' => 3]),
+        ]);
+
+        $this->assertMenuEquals(['Third', 'Second', 'First'], [
+            new MarkdownPage('first', ['navigation.priority' => 3]),
+            new MarkdownPage('second', ['navigation.priority' => 2]),
+            new MarkdownPage('third', ['navigation.priority' => 1]),
+        ]);
+    }
+
+    public function testMainNavigationMenuWithFrontMatterOrder()
+    {
+        $this->assertMenuEquals(['First', 'Second', 'Third'], [
+            new MarkdownPage('first', ['navigation.order' => 1]),
+            new MarkdownPage('second', ['navigation.order' => 2]),
+            new MarkdownPage('third', ['navigation.order' => 3]),
+        ]);
+
+        $this->assertMenuEquals(['Third', 'Second', 'First'], [
+            new MarkdownPage('first', ['navigation.order' => 3]),
+            new MarkdownPage('second', ['navigation.order' => 2]),
+            new MarkdownPage('third', ['navigation.order' => 1]),
+        ]);
+    }
+
+    public function testMainNavigationMenuWithFrontMatterLabel()
+    {
+        $this->assertMenuEquals(['First', 'Second', 'Third'], [
+            new MarkdownPage('foo', ['navigation.label' => 'First']),
+            new MarkdownPage('bar', ['navigation.label' => 'Second']),
+            new MarkdownPage('baz', ['navigation.label' => 'Third']),
+        ]);
+    }
+
+    public function testMainNavigationMenuWithFrontMatterHidden()
+    {
+        $this->assertMenuEquals(['Foo', 'Bar', 'Baz'], [
+            new MarkdownPage('foo', ['navigation.hidden' => false]),
+            new MarkdownPage('bar', ['navigation.hidden' => false]),
+            new MarkdownPage('baz', ['navigation.hidden' => false]),
+        ]);
+
+        $this->assertMenuEquals([], [
+            new MarkdownPage('foo', ['navigation.hidden' => true]),
+            new MarkdownPage('bar', ['navigation.hidden' => true]),
+            new MarkdownPage('baz', ['navigation.hidden' => true]),
+        ]);
+    }
+
+    public function testMainNavigationMenuWithFrontMatterVisible()
+    {
+        $this->assertMenuEquals(['Foo', 'Bar', 'Baz'], [
+            new MarkdownPage('foo', ['navigation.visible' => true]),
+            new MarkdownPage('bar', ['navigation.visible' => true]),
+            new MarkdownPage('baz', ['navigation.visible' => true]),
+        ]);
+
+        $this->assertMenuEquals([], [
+            new MarkdownPage('foo', ['navigation.visible' => false]),
+            new MarkdownPage('bar', ['navigation.visible' => false]),
+            new MarkdownPage('baz', ['navigation.visible' => false]),
+        ]);
+    }
+
+    public function testMainNavigationMenuWithFrontMatterGroup()
+    {
+        // TODO: For new v2 system, this should insert a root item with the group name and the children as the pages
```

Comment:
For new v2 system, this should insert a root item with the group name and the children as the pages